setup/cli-install.php

```diff
@@ -0,0 +1,172 @@
+ * php www/setup/install.php --database=my_db --database_user=db_user_12 --language=ru
+ */
+
+if (PHP_SAPI != 'cli') {
+    exit('This file should be run through the command line interface!');
+}
+$path = dirname(__FILE__) . '/';
+$config = $path . 'config.xml';
+$mode = 'new';
+
+$variables = [
+    'database_type' => 'mysql',
+    'database_server' => [
+        'prompt' => 'Database server',
+        'default' => '127.0.0.1',
+    ],
+    'database' => [
+        'prompt' => 'Database',
+        'default' => '',
+    ],
+    'database_user' => [
+        'prompt' => 'Database user',
+        'default' => '',
+    ],
+    'database_password' => [
+        'prompt' => 'Database password',
+        'default' => '',
+    ],
+    'database_connection_charset' => 'utf8',
+    'database_charset' => 'utf8',
+    'database_collation' => 'utf8_general_ci',
+    'table_prefix' => [
+        'prompt' => 'Table prefix',
+        'default' => 'modx_',
+    ],
+    'https_port' => 443,
+    'http_host' => [
+        'prompt' => 'Http host',
+        'default' => '',
+    ],
+    'cache_disabled' => 0,
+    'inplace' => 0,
+    'unpacked' => 0,
+    'language' => [
+        'prompt' => 'Language',
+        'default' => 'en',
+    ],
+    'cmsadmin' => [
+        'prompt' => 'Admin login',
+        'default' => 'admin',
+    ],
+    'cmspassword' => [
+        'prompt' => 'Admin password',
+        'default' => '',
+    ],
+    'cmsadminemail' => [
+        'prompt' => 'Admin email',
+        'default' => '',
+    ],
+    'core_path' => [
+        'prompt' => 'Path to core',
+        'default' => dirname($path) . '/core/',
+    ],
+    'context_mgr_path' => [
+        'prompt' => 'Path to manager',
+        'default' => dirname($path) . '/manager/',
+    ],
+    'context_mgr_url' => [
+        'prompt' => 'Manager url',
+        'default' => '/manager/',
+    ],
+    'context_connectors_path' => [
+        'prompt' => 'Path to connectors',
+        'default' => dirname($path) . '/connectors/',
+    ],
+    'context_connectors_url' => [
+        'prompt' => 'Url to connectors',
+        'default' => '/connectors/',
+    ],
+    'context_web_path' => dirname($path) . '/',
+    'context_web_url' => '/',
+    'remove_setup_directory' => 1,
+];
+
+// Parse CLI arguments
+$args = array_slice($argv, 1);
+$cli_variables = [];
+foreach ($args as $arg) {
+    $tmp = array_map('trim', explode('=', $arg));
+    if (count($tmp) === 2) {
+        $k = ltrim($tmp[0], '-');
+        if (isset($variables[$k])) {
+            $cli_variables[$k] = $tmp[1];
+        } elseif ($k == 'mode' && in_array($tmp[1], ['new', 'upgrade', 'advanced'])) {
+            $mode = $tmp[1];
+        }
+    }
+}
+
+// If it is upgrade - parse old variables
+$old_variables = [];
+if (file_exists($config)) {
+    $use = ($mode == 'upgrade')
+        ? 'Y'
+        : readline("There is an old config found at {$config}. Do you want to load this values? (Y): ");
```

Comment:
This is nitpicking, but I'd suggest `existing config file` instead of `old config` - old implies it's out of date. 

It looks like this will grab only from a config.xml in the root - not from a config.inc.php if it exists. Not sure if it has to, but what do you think about that for upgrades? Or is this not meant to be used with upgrades?

Reply:
It is not for upgrade, it for continue current install after abort. And it would get file from /setup/ directory

To upgrade you need specify `--mode=upgrade` and if there will be no `config.xml`, just answer again to questions. Maybe we can load existing config, really.

Reply:
Do'nt know if it is your case, but we are using servers with plesk having multiple php version, and need to define the path to php command as well depending on the version used by the modx installation
Current example of command line installation /upgrade  :  /opt/plesk/php/7.1/bin/php /var/www/vhosts/domain.tld/httpdocs/setup/index.php --installmode=upgrade-advanced --core_path=/var/www/vhosts/domain.tld/core/ --config=domain.tld.config.xml;

Reply:
You mean, that if just type `php` in console it will answer `command not found`? There are always should be default `php`.

Please, check it in your environment.

Reply:
At least one big german hoster that has php 4.4.3 installed as default php in the console. For whatever reason.

Reply:
That's not something this PR has to address. There are different ways to make the desired binary available as `php`.

Reply:
It should be safe to assume the `php` in `$PATH` is a correct PHP version that is above the minimal requirement. This PR should not have to deal with any such edge cases.

Reply:
Vasilly, There is a php, but it is a php5.6, old stuff, I prefer to install all websites with the most performance wyse version of it...

Reply:
@OC, not sure the PATH would be something I will consider as granted... In that case, it is not valid in ssh command line gives you php old version

Reply:
Most applications assume this. I have never seen any command line tool that goes into great detail to explain `PATH` and covering weird edge cases. That is up to the developer to fix the environment. See [composer](https://getcomposer.org/doc/00-intro.md#locally) for instance, their installation instructions are just

```
php composer-setup.php --install-dir=bin --filename=composer 
```

That YOUR `php` executable gives you incorrect version is not our fault, it is your configuration that is incorrect. On my server `php` gives me the latest version I have installed.

So yeah, in the world of command line tools, you would assume that the `PATH` and related settings/configurations are correct.

Additionally, there is no way to cover _all_ the edge cases in the world for this scenario.

Reply:
Solved the problem differently - index.php is now simply included inside the script

Reply:
Optimus, This is not our php version, but as Jakko said, there are lost of sutiation were php with the valid version is not active. Plesk users and there are thousands will tell you it's the case.
Regarding composer, we can do /opt/plesk/php/7.1/bin/php composer-setup.php --install-dir=bin --filename=composer and its done because from what I know, it does not call another php command inside .
The issue arise when the php command is called inside a command file and thus must be configurable in some way for such users.

Reply:
@AmaZili problem is already solved https://github.com/modxcms/revolution/pull/13790/commits/12db2266379a01e7fc46e096e10d14337b80ea55#diff-bdb40b842e5d2c02e8822780192e0d72R183